### ISSUE TITLE
[Bug]: esp32 DevkitC v4 Looses Connection

### ISSUE THREAD

**Issue by Nubi123bty** [openhaptics/openhaptics-firmware#32]
For some reason, the Windows Client is unable to connect to the ESP, as soon as it is found and an attempt to connect is made it loses the connection to the board over BT. 
this is the error created by the player:

```
2022-11-28 15:47:40,361 [1] ERROR BhapticsPlayer.Internal.BhapticsLiteDB: BhapticsDB DB created
2022-11-28 15:47:40,438 [1] ERROR BhapticsPlayer.Utils.Utilities: Util Get BhapticsSettingValue 1
2022-11-28 15:47:40,439 [1] ERROR BhapticsPlayer.Utils.Utilities: CurrentCulture: de-AT
2022-11-28 15:47:40,439 [1] ERROR BhapticsPlayer.Utils.Utilities: Properties.Settings.Default.language Before: en-US
2022-11-28 15:47:40,439 [1] ERROR BhapticsPlayer.Utils.Utilities: Properties.Settings.Default.language: en-US
2022-11-28 15:47:40,731 [1] WARN  BhapticsPlayer.Internal.SettingsStore: CheckDfuFailure
2022-11-28 15:47:40,840 [1] ERROR BhapticsPlayer.Utils.AudioToHapticConfig: Load default setting
2022-11-28 15:47:41,014 [1] ERROR BhapticsPlayer.Audio.Utils.AudioDeviceWatcher: AudioDeviceWatcher() default selected. Die Sequenz enthält kein übereinstimmendes Element.
2022-11-28 15:47:41,131 [1] ERROR BhapticsPlayer.WebSocket.StudioReceiver: Listening on port 15881, and providing WebSocket services:
2022-11-28 15:47:41,145 [1] ERROR BhapticsPlayer.WebSocket.StudioReceiver: Listening on port 15882, and providing WebSocket Secure services: True
2022-11-28 15:47:41,145 [1] WARN  BhapticsPlayer.Utils.Utilities: Release Id : 2009
2022-11-28 15:47:41,145 [1] WARN  BhapticsPlayer.Utils.Utilities: releaseNum : 2009
2022-11-28 15:47:41,655 [1] WARN  BhapticsPlayer.Internal.SettingsStore: CheckDfuFailure
2022-11-28 15:47:41,745 [5] ERROR BhapticsModLauncher.Config.BhapticsContentsRepository: HttpWebResponse status: OK
2022-11-28 15:47:42,448 [5] ERROR BhapticsModLauncher.Config.BhapticsContentsRepository: BhapticsContentsRepository Exists DB
2022-11-28 15:47:42,486 [5] ERROR BhapticsModLauncher.Config.BhapticsContentsRepository: BhapticsContentsRepository SaveGameContents()
2022-11-28 15:47:52,765 [6] WARN  BhapticsPlayer.Bluetooth.DeviceWatcherService: Remove Device TactosyDevice {Name=TactSuitX40, Id=70518412007866, Paired=False, Connected=Disconnected}
2022-11-28 15:48:00,282 [30] WARN  BhapticsPlayer.Bluetooth.DeviceWatcherService: Remove Device TactosyDevice {Name=TactSuitX40, Id=70518412007866, Paired=False, Connected=Disconnected}
```

Serial output does not produce anything bt this:
```
>>      onConnect()
>>      onConnect(*param)
>>      onDisconnect()
```

**Comment by leon0399** [comment]
Thank you for submitting an issue! Please, help me investigate your issue. I'll need more detailed info, such as

- which version/branch of firmware you are using 
- what device you are building
- bHaptics Player version (Can be found in Settings menu)

Also, please, try switching to `v0.1.2` version with following steps: 
![Code_RRV9HMU6qJ](https://user-images.githubusercontent.com/1759654/204336458-e9b10db9-4907-4ba0-9614-fa70bbb85825.gif)

**Comment by Nubi123bty** [comment]
thank you for looking into it, 
Building the tactx40, but having the same issue with any other Build on the devkitC V4, Tried 4 different Boards, I am using the 1.2 as recommended, the Player version is 1.8.2.2, had the same issue with the previous release, 
the connection is breaking off several times until after 10 to 20 minutes there seems to be a connection.
this is the final serial output on the serial port when connected.
i also tried to downgrade ESP32 Library version to 2.0.0 while testing, with the same issue. 



`>>      onConnect()
>>      onConnect(*param)
>>      onDisconnect()
>>      onConnect()
>>      onConnect(*param)
>>      onDisconnect()
>>      onConnect()
>>      onConnect(*param)
>>      onDisconnect()
>>      onConnect()
>>      onConnect(*param)
>>      onDisconnect()
>>      onConnect()
>>      onConnect(*param)
>>      onDisconnect()
>>      onConnect()
>>      onConnect(*param)
>>      onDisconnect()
>>      onConnect()
>>      onConnect(*param)
>>      onDisconnect()
>>      onConnect()
>>      onConnect(*param)
>>      onDisconnect()
>>      onConnect()
>>      onConnect(*param)
>>      onDisconnect()
>>      onConnect()
>>      onConnect(*param)
>>      onDisconnect()
>>      onConnect()
>>      onConnect(*param)
>>      onDisconnect()
>>      onConnect()
>>      onConnect(*param)
>>      onDisconnect()
>>      onConnect()
>>      onConnect(*param)
>>      onRead (UUID: 6e400007-b5a3-f393-e0a9-e50e24dcca9e) 
        value: `␂␁`, len: 2
>>      onRead (UUID: 6e400008-b5a3-f393-e0a9-e50e24dcca9e) 
        value: `d`, len: 2
>>      onRead (UUID: 6e400003-b5a3-f393-e0a9-e50e24dcca9e) 
        value: `$��R=f)���`, len: 10
>>      onRead (UUID: 6e40000b-b5a3-f393-e0a9-e50e24dcca9e) 
        value: ``, len: 2`

versions of VS:
Version: 1.73.1 (user setup)
Commit: 6261075646f055b99068d3688932416f2346dd3b
Datum: 2022-11-09T04:27:29.066Z
Electron: 19.0.17
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
Betriebssystem: Windows_NT x64 10.0.19045
Sandkasten: No

**Comment by leon0399** [comment]
Have you tried connecting to ESP from your phone? Or Meta Quest of you have one

I have an idea why it might happening, I'll lol into it tomorrow

DFU is responsible for firmware update, and presumably bhaptics player trying to update firmware (?). I'll try bumping version to latest one and see how it performs

**Comment by Nubi123bty** [comment]
the mobile device paired right away.

**Comment by leon0399** [comment]
I've bumped the firmware version to prevent it from trying to update, let's try a new version. Don't forget to checkout the latest master branch

**Comment by Nubi123bty** [comment]
no, still the same problem, only after many attempts is the connection established, I just recognized that the Second PWM on address 0x41 is not getting any signals, does it have to do something with this issue?

**Comment by Nubi123bty** [comment]
yes, I have, the first bit switched 3 boards and checked the boards with the sample code.  I remember in an earlier version Serial output did give a success or fail message on finding the boards.

**Comment by Nubi123bty** [comment]
please ignore the problem with the PWM, apparently, the cable was broken, showing contact enough for phantom voltage but no current coming through.

**Comment by Nubi123bty** [comment]
ok one thing I was able to test the connection on Oculus Quest 2 (SW-Ver.: 415550688) Bhaptics Oculus Player Ver.: 0.7.14, Firmware on ESP Tact X40 V1.2.
I can't connect with the Tact X40, The Gloves I made on the ESP32 Devkitv1 Work fine. no error message is produced.

**Comment by leon0399** [comment]
> The Gloves I made on the ESP32 Devkitv1 Work fine

By gloves, you mean "Tactosy for hands" compatible device?

**Comment by leon0399** [comment]
I've just verified that it works correctly on the latest `master` branch, bHaptics Player 1.8.2.2
Consider joining our Discord, we've been discussing possible causes of this problem, one of them is the bad quality of either the Bluetooth adapter/dongle or the ESP chip itself and how to distinguish good quality ESPs from bad ones

![image](https://user-images.githubusercontent.com/1759654/205345014-4aa2c069-4493-4377-8111-fc35f22c2941.png)

**Comment by Nubi123bty** [comment]
Thank you very much, for your time looking into it. I will try to get another, than from AZ-Delivery, to try it out.

**Comment by Nubi123bty** [comment]
is there a possibility that the BT draws too much Power from the esp when connecting? what version of the ESP itself are you using or what brand is the Devkit?

**Comment by Akluma** [comment]
I have same problem.
